### DISCUSSION TITLE
Get current path in a reactive manner?

### DISCUSSION THREAD

**Discussion by mrlubos** [TanStack/router#1075] [Q&A] [answered]
With `react-router`, I can use `location` to send data to analytics

```tsx
import { Outlet, useLocation } from 'react-router-dom';

function RouteWrapper() {
  const location = useLocation();

  return (
    <>
      <Analytics pathname={location.pathname} />
      <Outlet />
    </>
  );
}
```

How can I achieve the same with this package?

```tsx
import { Outlet, RootRoute } from '@tanstack/react-router'

function RouteWrapper() {
  const location = null // ???

  return (
    <>
      <Analytics pathname={location.pathname} />
      <Outlet />
    </>
  )
}

export const Route = new RootRoute({
  component: RouteWrapper,
})
```

**Comment by borstessi** [comment, marked as answer]
@mrlubos 

i think i found the solution.

you can use the useRouterState Hook:

```
const router = useRouterState();

  console.log("here", router.location.pathname);
```

it is reactive and will change on navigation

**Reply by Emmanuel-Xs** [reply]
what about  a from path i.e a path from where we got to the current path

**Reply by mrlubos** [reply]
Just want to update here, TanStack router now has `useLocation()` hook too.

**Reply by Emmanuel-Xs** [reply]
Nice, thanks

**Reply by Akronae** [reply]
wait, but those paths aren't type safe 😕

**Reply by TrawenCZ** [reply]
> wait, but those paths aren't type safe 😕

I think that's because there's no guarantee that user wouldn't just navigate to random URL via rewriting the URL or clicking some external link. Thus, the paths can be whatever.
